sourmash/sbt.py

```diff
@@ -279,13 +292,17 @@ def search(self, query, *args, **kwargs):
         best_only = kwargs['best_only']
         unload_data = kwargs.get('unload_data', False)
 
+        # If True, returns the leaf node (rather than the signature data) and the
+        # position of the leaf on the tree
+        return_leaf = kwargs.get('return_leaf', False)
```

Comment:
you know this code base better than me, but if you're querying the kwargs a lot like this do you think this whole function might be better adapted to a class? Maybe named `Finder` or something like that? That way you could also break up this function a little more and create methods for searching, finding scaled value of the tree, creating a search queue, etc?